### PR TITLE
[Integration]: Whatsapp Integration

### LINKED ISSUE #8050: [Integration] Integrate Whatsapp with MindsDB
The goal of this issue is to build a new handler that will utilize the WhatsApp API to connect MindsDB to WhatsApp.

## Resources :bulb: 

:checkered_flag: To get started, please refer to the following resources:

* :books:  [Building Application Handler docs](https://docs.mindsdb.com/contribute/app-handlers): This documentation will guide you through the process of building a custom handler for MindsDB. It contains essential information on the handler structure, methods, and best practices.

* :book:  [WhatsApp docs](https://developers.facebook.com/docs/whatsapp/)

* :point_right: [Twitter Handler](https://github.com/mindsdb/mindsdb/tree/staging/mindsdb/integrations/handlers/twitter_handler): This existing handler connects MindsDB with the Twitter API, allowing users to automate tasks related to posting and analyzing tweets. You can use the Twitter Handler as a reference for building the Slack API integration.

*  :point_right:  [Binance API Handler](https://github.com/mindsdb/mindsdb/tree/staging/mindsdb/integrations/handlers/binance_handler): Another example of a successful MindsDB integration to the Binance API to make trade data available to use for model training and predictions.

## Next Steps :male_detective: :female_detective: 
 
-   You can familiarize yourself with the WhatsApp API.
-   Create a new handler that connects MindsDB to WhatsApp.
-   Implement the core features mentioned above.
-   Thoroughly test the integration to ensure proper functionality.
-   Submit a pull request with the new WhatsApp Integration handler.

## :loudspeaker: Additional rewards :trophy:

Each App integration brings :five: :zero:  : points for SWAG and entry into the draw for a  :point_down: :
-  :computer:  [Razer Blade 16 Laptop](https://www.razer.com/gaming-laptops/Razer-Blade-16/RZ09-0483SEJ3-R3U1)
- :shirt: :bear:  [MindsDB Swag](https://store.covver.io/mindsdb/collections/swag-collection)
:information_source: For more info check out https://mindsdb.com/hacktoberfest/ :point_left:

### PR DESCRIPTION
**Pull request by tcchawla**
## Description

This PR implements a Whatsapp Handler, through which users can send messages and reply to the messages using MindsDB.

Fixes #8050 #6990 #5352 

## Type of change

- [x] ⚡ New feature (non-breaking change which adds functionality)

## Verification Process

To ensure the changes are working as expected:

 - [ ]   Test Location: Specify the URL or path for testing.
 - [x]   Verification Steps: Outline the steps or queries needed to validate the change. Include any data, configurations, or actions required to reproduce or see the new functionality.

## Queries to Verify

```sql
-- whastapp testing
CREATE DATABASE whatsapp_test
WITH ENGINE = "whatsapp",
PARAMETERS = {
  "account_sid": "AC37910d1a308d0c53d1dc9b7f1f352a34",
  "auth_token": "5ab63e11d89aec875d5a212af1e7223e"
  };

-- Get all messages 
SELECT * FROM whatsapp_test.messages LIMIT 100;

-- Send a whatsapp message
INSERT INTO whatsapp_test.messages (body, from_number, to_number)
VALUES('woww', 'whatsapp:+14155238886', 'whatsapp:+15519990863');

SELECT * FROM whatsapp_test.messages where sid="SM375f075778f91b56634ce5d92db249cd"

SELECT * FROM whatsapp_test.messages where from_number="whatsapp:+15519990863";
```

## Additional Media:

- [x] I have attached a brief loom video or screenshots showcasing the new functionality or change.

Screenshot 2023-11-24 at 10 38 30 PM

Screenshot 2023-11-24 at 10 45 03 PM

Screenshot 2023-11-24 at 10 40 48 PM

Screenshot 2023-11-24 at 10 42 45 PM

## Checklist:

- [x] My code follows the style guidelines(PEP 8) of MindsDB.
- [x] I have appropriately commented on my code, especially in complex areas.
- [x] Necessary documentation updates are either made or tracked in issues.
- [x] Relevant unit and integration tests are updated or added.

### CODE REVIEW

File: mindsdb/integrations/handlers/whatsapp_handler/whatsapp_handler.py [hunk 1]
```diff
@@ -0,0 +1,406 @@
...
+        self._register_table('messages', messages)
+
+    def connect(self):
+        """
+        Authenticate with the Twilio API using the provided `account_SID` and `auth_token`.
+        """
+        if self.is_connected is True:
+            return self.client
+
+        self.client = Client(
```
Comment (ZoranPandovski) [inline mindsdb/integrations/handlers/whatsapp_handler/whatsapp_handler.py:209]: Is the number required here?

File: mindsdb/integrations/handlers/whatsapp_handler/whatsapp_handler.py [hunk 2]
```diff
@@ -0,0 +1,406 @@
...
+
+            # Check text length
+            max_text_len = 1500
+            text = params["body"]
+            words = re.split('( )', text)
+            messages = []
+
+            # Check text pattern
+            text2 = ''
+            pattern = r'http[s]?://(?:[a-zA-Z]|[0-9]|[$-_@.&+]|[!*\(\),]|(?:%[0-9a-fA-F][0-9a-fA-F]))+'
```
Comment (ZoranPandovski) [inline mindsdb/integrations/handlers/whatsapp_handler/whatsapp_handler.py:134]: Do we need to replace urls? If so can we better comment this so we know what happens here

**Review by ZoranPandovski — APPROVED** [review]
Thanks. I've added Two minor comments

File: mindsdb/integrations/handlers/whatsapp_handler/whatsapp_handler.py (hunk 2 above)
Comment (tcchawla) [reply mindsdb/integrations/handlers/whatsapp_handler/whatsapp_handler.py:145]: I updated the comments.

**Review by ZoranPandovski — APPROVED** [review]

**Pull request merged**